clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2591,6 +2544,10 @@ class SyclKernelBodyCreator : public SyclKernelFieldHandler {
 
     const auto *RecordDecl = Ty->getAsCXXRecordDecl();
     createSpecialMethodCall(RecordDecl, getInitMethodName(), BodyStmts);
+    CXXMethodDecl *FinalizeMethod =
+        getMethodByName(RecordDecl, FinalizeMethodName);
+    if (FinalizeMethod)
+      createSpecialMethodCall(RecordDecl, FinalizeMethodName, FinalizeStmts);
```

Comment:
It is my understanding that stream classes are expected to have a finalize method.

Reply:
I'm more wondering whether we should generate an error if we can't find a finalize method so that the library cannot accidentally add a stream without a finalize method and have it silently do whatever.